src/billing/context/billing.tsx

```diff
@@ -139,24 +151,46 @@ export const BillingProvider: FC<Props> = React.memo(({children}) => {
   }, [dispatch])
 
   const handleUpdatePaymentMethod = useCallback(
-    async (_paymentMethodId: string) => {
+    async (paymentMethodId: string) => {
       try {
-        const response = await getPaymentForm({form: 'billing'})
-        // const response = await putBillingPaymentMethod({data: paymentMethodId})
+        const resp = await putBillingPaymentMethod({
+          data: {paymentMethodId},
+        })
 
-        if (response.status !== 200) {
-          throw new Error(getErrorMessage(response))
+        if (resp.status !== 200) {
+          throw new Error(resp.data.message)
```

Comment:
what's the reasoning behind not using `getErrorMessage` here?

Reply:
We’re doing that in the catch statement later on.